parser/grammar.y

```diff
@@ -439,6 +439,15 @@ string_declarations
       }
     | string_declarations string_declaration
       {
+        new_ident := $2.GetIdentifier()
+        for _, ident := range $1 {
+          // Don't error if both identifiers are the empty string.
+          if old_ident := ident.GetIdentifier();  new_ident != "" && old_ident != "" && new_ident == old_ident {
+            return asLexer(yrlex).setError(
+              gyperror.DuplicateStringError,
+              `duplicate string identifier "%s"`, $2.GetIdentifier())
```

Comment:
It may be useful to include the name of the rule containing the duplicate string in the error message, but I think this implies that the validation must be done at the `rule` level, when the `ast.Rule` object is being constructed. An additional complication is that in order to report the line number where the duplicate string resides, instead of the line number where the rule ends, we may need to implement a variant of `setError` that accepts the line number as an argument, instead of using the current line number. We can pass the string's line number to that variant of set error while creating the error (the `ast.String` interface has a `GetLineNo` method)

Reply:
Good point! I think I've implemented it correctly but please tell me if there's a better way to do it as I'm still learning my way through the basics of golang. :)